src/group_theory/perm/cycle_type.lean

```diff
@@ -258,4 +259,70 @@ def partition (σ : perm α) : partition (fintype.card α) :=
 
 end partition
 
+/-!
+### 3-cycles
+-/
+
+/-- A three-cycle is a cycle of length 3. -/
+def is_three_cycle [decidable_eq α] (σ : perm α) : Prop := σ.cycle_type = {3}
+
+namespace is_three_cycle
+
+variables [decidable_eq α] {σ : perm α}
+
+lemma cycle_type (h : is_three_cycle σ) : σ.cycle_type = {3} := h
+
+lemma card_support (h : is_three_cycle σ) : σ.support.card = 3 :=
+by rw [←sum_cycle_type, h.cycle_type, singleton_eq_singleton, multiset.sum_cons, sum_zero]
+
+lemma _root_.card_support_eq_three_iff : σ.support.card = 3 ↔ σ.is_three_cycle :=
+begin
+  refine ⟨λ h, _, is_three_cycle.card_support⟩,
+  by_cases h0 : σ.cycle_type = 0,
+  { rw [←sum_cycle_type, h0, sum_zero] at h,
+    exact (ne_of_lt zero_lt_three h).elim },
+  obtain ⟨n, hn⟩ := exists_mem_of_ne_zero h0,
+  by_cases h1 : σ.cycle_type.erase n = 0,
+  { rw [←sum_cycle_type, ←cons_erase hn, h1, multiset.sum_singleton] at h,
+    rw [is_three_cycle, ←cons_erase hn, h1, h, singleton_eq_singleton] },
+  obtain ⟨m, hm⟩ := exists_mem_of_ne_zero h1,
+  rw [←sum_cycle_type, ←cons_erase hn, ←cons_erase hm, multiset.sum_cons, multiset.sum_cons] at h,
+  linarith [two_le_of_mem_cycle_type hn, two_le_of_mem_cycle_type (mem_of_mem_erase hm)],
+end
+
+lemma is_cycle (h : is_three_cycle σ) : is_cycle σ :=
+by rw [←card_cycle_type_eq_one, h.cycle_type, singleton_eq_singleton, card_singleton]
+
+lemma sign (h : is_three_cycle σ) : sign σ = 1 :=
+begin
+  rw [sign_of_cycle_type, h.cycle_type],
+  refl,
+end
+
+lemma inv {f : perm α} (h : is_three_cycle f) : is_three_cycle (f⁻¹) :=
+by rwa [is_three_cycle, cycle_type_inv]
```

Comment:
A simp lemma `inv_iff : is_three_cycle (f⁻¹) ↔ is_three_cycle f` might be a good addition, and should be easy to prove using this lemma.